performance_regression_cdc_test.py

```diff
@@ -0,0 +1,294 @@
+import pprint
+import time
+
+from concurrent.futures import ThreadPoolExecutor
+
+from sdcm.cluster import BaseNode, UnexpectedExit, Failure, Setup
+from sdcm.results_analyze import QueryFilter, QueryFilterCS, PerformanceResultsAnalyzer
+from performance_regression_test import PerformanceRegressionTest
+
+PP = pprint.PrettyPrinter(indent=2)
+
+
+class CDCQueryFilter(QueryFilter):
+    def filter_test_details(self):
+        test_details = 'test_details.job_name: \"{}\" '.format(
+            self.test_doc['_source']['test_details']['job_name'].split('/')[0])
+        test_details += self.test_cmd_details()
+        test_details += ' AND test_details.time_completed: {}'.format(self.date_re)
+        test_details += r' AND test_details.test_name: *cdc* '
+        return test_details
+
+    def test_cmd_details(self):
+        pass
+
+
+class CDCQueryFilterCS(QueryFilterCS, CDCQueryFilter):
+    def cs_params(self):
+        return self._PROFILE_PARAMS if 'profiles' in self.test_name else self._PARAMS
+
+
+class CDCPerformanceResultsAnalyzer(PerformanceResultsAnalyzer):
+    def _query_filter(self, test_doc, is_gce):
+        return CDCQueryFilterCS(test_doc, is_gce)()
+
+
+class PerformanceRegressionCDCTest(PerformanceRegressionTest):
+    keyspace = None
+    table = None
+
+    def test_write_with_cdc(self):
+        write_cmd = self.params.get("stress_cmd_w")
+
+        _ = self._workload_cdc(write_cmd, 2,  # pylint: disable=unused-variable
+                               test_name="test_write",
+                               sub_type="cdc_disabled")
+
+        node1: BaseNode = self.db_cluster.nodes[0]
+
+        with self.cql_connection_patient(node1) as session:
+            session.execute(f"TRUNCATE TABLE {self.keyspace}.{self.table}")
+            session.execute(f"ALTER TABLE {self.keyspace}.{self.table} WITH cdc = {{'enabled': true}}")
+
+        _ = self._workload_cdc(write_cmd, 2,  # pylint: disable=unused-variable
+                               test_name="test_write",
+                               sub_type="cdc_enabled")
+        self.check_regression_with_baseline(base_line="cdc_disabled")
+
+    def test_write_with_cdc_pre_image(self):
+        write_cmd = self.params.get("stress_cmd_w")
+
+        _ = self._workload_cdc(write_cmd, 2,  # pylint: disable=unused-variable
+                               test_name="test_write",
+                               sub_type="cdc_disabled")
+        node1: BaseNode = self.db_cluster.nodes[0]
+
+        with self.cql_connection_patient(node1) as session:
+            session.execute(f"TRUNCATE TABLE {self.keyspace}.{self.table}")
+            session.execute(f"TRUNCATE TABLE {self.keyspace}.{self.table}_scylla_cdc_log")
+            session.execute(
+                f"ALTER TABLE {self.keyspace}.{self.table} WITH cdc = {{'enabled': true, 'preimage': true}}")
```

Comment:
Do we have plan to test performance effect of different ttl?

Reply:
This should be discussed.